frontend/kesaseteli/employer/src/hooks/backend/useOperationPermitted.ts

```diff
@@ -0,0 +1,12 @@
+import useLogoutQuery from 'kesaseteli/employer/hooks/backend/useLogoutQuery';
+import useAuth from 'shared/hooks/useAuth';
+
+const useIsOperationPermitted = (): boolean => {
+  const { isAuthenticated } = useAuth();
+  const {
+    isLoading: isLoadingLogout,
+    isSuccess: isLoggedOut,
+  } = useLogoutQuery();
+  return isAuthenticated && !isLoadingLogout && !isLoggedOut;
```

Comment:
Is there a way that the user would be logged out AND authenticated?! I think `!isLoggedOut` seems redundant.

Reply:
I'm not sure if it's completely same. I can rename `isLoggedOut` to `isLogoutSucceeded`. There might be a tiny moment between the logout response returns successful result and `isAuthenticated` is false. The latter is evaluated after some authorized content is rendered and `withAuth` hoc has called backend to retrieve user data and response is 403.

The reason I added this hook when I realized that there will be some extra page renders with extra backend calls while clicking the logout button (that also caused lots of problems with component tests). Other way to tackle the problem is to somehow prevent extra renders when clicking logout button.

Reply:
Also I dont know what happens if logout responses error, is user still authenticated or not 🤷